diffpy/snmf/stretchednmfapp.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import argparse
+
+from diffpy.snmf.io import load_input_signals, initialize_variables
+from diffpy.snmf.subroutines import update_weights_matrix, get_residual_matrix, objective_function
+
+
+def create_parser():
+    parser = argparse.ArgumentParser(
+        prog="stretched_nmf",
+        description="Stretched Nonnegative Matrix Factorization"
+    )
+    parser.add_argument('-v', '--version', action='version', help='Print the software version number')
+    parser.add_argument('-i', '--input-directory', type=str, default=None,
+                        help="Directory containing experimental data. Default before will cause the program to use the current working directory as the input directory.")
+    parser.add_argument('-o', '--output-directory', type=str,
+                        help="The directory where the results will be dumped. Default behavior will create a new directory named 'smnf_results' inside the input directory.")
+    parser.add_argument('-t', '--data-type', type=str, choices=['xrd', 'pdf'],
+                        help="The type of the experimental data.")
+    parser.add_argument('components', type=int,
+                        help="The number of component signals to obtain from experimental "
+                             "data. Must be an integer greater than 0.")
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = create_parser()
+
+    grid, data_input, data_type = load_input_signals(args.input_directory)
+    if args.data_type is not None:
+        variables = initialize_variables(data_input, args.components, args.data_type)
+    else:
+        variables = initialize_variables(data_input, args.components, data_type)
+
+    if variables["data_type"] == 'pdf':
+        lifted_data = data_input - np.ndarray.min(data_input[:])
+
+    weights_matrix = variables["weight_matrix_guess"]
+    component_matrix = variables["component_matrix_guess"]
+    stretching_factor_matrix = variables["stretching_matrix_guess"]
```

Comment:
you did this for half the variables and not the others.  I would just not do this and get the values from the dict below.

If you lint the code with black (whatever you do, do that on a separate branch! it makes the code review impossible to mix linting edits with functional edits!) it will arrange this in a nice readable way.

Reply:
I understand, I will remove this. 

I will look into setting up black to make my code more readable and better formatted.